### DISCUSSION TITLE
Unable to run Mac tests

### DISCUSSION THREAD

**Discussion by tciuro** [community/community#25189] [Actions]
<p>I can run the tests locally using Xcode and xcodebuild, but when GitHub CI runs it, it shows the following err:</p>
<pre><code>No signing certificate "Mac Development" found: No "Mac Development" signing certificate matching team ID "HPSB123WH5" with a private key was found.
</code></pre>
<p>I’m not sure how to setup the project so that GitHub can run the tests. I was hoping to find a “Do not Code Sign” option in Build Settings, but the Code Sign section only allows signing options. Any ideas?</p>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/tciuro">@tciuro</a> There is <a href="https://github.com/marketplace/actions/import-code-signing-certificates" rel="nofollow noopener">an action</a> which could import code-signing certificates, you need to export a .p12 file with password and convert the .p12 file it to a base64 string, then store the password and base64 string in the scecrets.  Could this action help? </p>

**Comment by tciuro** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> Thanks for the link! Sounds really interesting, it was easy to export the p.12 file and convertt it base64. However, I have a couple questions which I filed here: <a href="https://github.com/Apple-Actions/import-codesign-certs/issues/6" rel="nofollow noopener">https://github.com/Apple-Actions/import-codesign-certs/issues/6</a></p>
<p>Do you happen to know how to configure it?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/tciuro">@tciuro</a> , Import Code-Signing Certificates is an github action , you could add it to your github actions <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#creating-a-workflow-file" rel="nofollow noopener">workflow yml file</a>. </p>
<p>The two attributes <strong>p12-file-base64</strong> and <strong>p12-password</strong> are not referring to file , you need to store the base64 file content and your password into secrets. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/24cbfb7aeb34723768921c45993dd1f66ded9807.png" data-download-href="/uploads/short-url/5fwdAc52WmW9bvPJEzYFADMkbIP.png?dl=1" title="secrets.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181089319-ecd54054-0b60-4eef-9fc5-8f308f0aad59.png" alt="secrets.png" data-base62-sha1="5fwdAc52WmW9bvPJEzYFADMkbIP" width="690" height="395" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/24cbfb7aeb34723768921c45993dd1f66ded9807_2_690x395.png, https://aws1.discourse-cdn.com/github/original/2X/2/24cbfb7aeb34723768921c45993dd1f66ded9807.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/24cbfb7aeb34723768921c45993dd1f66ded9807.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/24cbfb7aeb34723768921c45993dd1f66ded9807_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">secrets.png</span><span class="informations">1034×593 30.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>There is the document for how to use secrets: <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets." rel="nofollow noopener">https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets. </a></p>

**Comment by tciuro** [comment]
<p>Thank you very much!!! :+1:t2::pray:t2:</p>
